### ISSUE TITLE
When to autoplay room audio?

### ISSUE THREAD

**Issue by bd339** [Senscape/Dagon#157]
Today I had a discussion with @civanT about the behaviour of room audio. He was using room audio for background music, and at some point he stopped the room audio and started another audio in its stead and then left the room. Upon returning to the room, both room audio were playing, because that's just the way room audio is designed.

Of course it is possible for us to change that behaviour, so that if a room audio was manually stopped it wouldn't autoplay any more. But that adds a complexity to the concept of room audio. Someone might then ask: What if I manually start a room audio that was manually stopped? Should it then revert to its autoplaying behaviour?

One way to deal with the current behaviour is to think of room audio as: audio that should always play when in a specific room. That would mean developers should use the room enter/leave events to manually handle conditional room audio (if condition X, play Happy else play Scary).

On the one hand, it is nice if Dagon automatically handles room audio in a way the developer expects, but on the other hand, it might be better to just keep room audio simple and let the developer handle their specific needs in the room callbacks? Suggestions and opinions are welcome.

**Comment by agustincordes** [comment]
Well, thing is, Room Audio was never designed for music but rather ambient audio that is always going to be present. It's not very flexible but always worked for its purpose. My advice in this case would be to use separate audio tracks that can be referenced and played/stopped at will.

**Comment by bd339** [comment]
I agree. There is not much incentive to change the behaviour when developers can already achieve whatever behaviour they desire by restricting their use of room audio to audio that is always present.
This wasn't intended as a bug report as I don't consider the current behaviour buggy. Just wanted to document this for others who might stumble into the problem and find themselves in need of ideas to tackle the problem. Also on the off chance that someone reads it and has a brilliant suggestion :smile:
